homeserver_dendrite.go

```diff
@@ -0,0 +1,164 @@
+		r30_users_ios BIGINT,
+		r30_users_electron BIGINT,
+		r30_users_web BIGINT,
+		r30v2_users_all BIGINT,
+		r30v2_users_android BIGINT,
+		r30v2_users_ios BIGINT,
+		r30v2_users_electron BIGINT,
+		r30v2_users_web BIGINT,
+		cpu_average BIGINT,
+		memory_rss BIGINT,
+		user_agent TEXT,
+		daily_user_type_native BIGINT,
+		daily_user_type_bridged BIGINT,
+		daily_user_type_guest BIGINT,
+		database_engine TEXT,
+		database_server_version TEXT,
+		log_level TEXT,
+		goos TEXT,
+		goarch TEXT,
+		goversion TEXT,
+		federation_disabled INT,
+		monolith INT,
+		nats_embedded INT,
+		nats_in_memory INT,
+		num_cpu INT,
+		num_go_routine INT,
+		version TEXT
+		)`)
+
+	return err
+}
+
+func (sr *ReportStatsDendrite) Save(db *sql.DB) error {
+	cols := []string{"homeserver", "local_timestamp", "remote_addr"}
+	vals := []interface{}{sr.Common.Homeserver, sr.Common.LocalTimestamp, sr.Common.RemoteAddr}
+
+	cols, vals = appendIfNonNil(cols, vals, "remote_timestamp", sr.Common.RemoteTimestamp)
+	cols, vals = appendIfNonNil(cols, vals, "uptime_seconds", sr.Common.UptimeSeconds)
+	cols, vals = appendIfNonNil(cols, vals, "total_users", sr.Common.TotalUsers)
+	cols, vals = appendIfNonNil(cols, vals, "total_nonbridged_users", sr.Common.TotalNonBridgedUsers)
+	cols, vals = appendIfNonNil(cols, vals, "total_room_count", sr.Common.TotalRoomCount)
+	cols, vals = appendIfNonNil(cols, vals, "daily_active_users", sr.Common.DailyActiveUsers)
+	cols, vals = appendIfNonNil(cols, vals, "daily_active_rooms", sr.Common.DailyActiveRooms)
+	cols, vals = appendIfNonNil(cols, vals, "daily_messages", sr.Common.DailyMessages)
+	cols, vals = appendIfNonNil(cols, vals, "daily_sent_messages", sr.Common.DailySentMessages)
+	cols, vals = appendIfNonNil(cols, vals, "daily_active_e2ee_rooms", sr.Common.DailyActiveE2eeRooms)
+	cols, vals = appendIfNonNil(cols, vals, "daily_e2ee_messages", sr.Common.DailyE2eeMessages)
+	cols, vals = appendIfNonNil(cols, vals, "daily_sent_e2ee_messages", sr.Common.DailySentE2eeMessages)
+	cols, vals = appendIfNonNil(cols, vals, "monthly_active_users", sr.Common.MonthlyActiveUsers)
+
+	cols, vals = appendIfNonNil(cols, vals, "r30_users_all", sr.Common.R30UsersAll)
+	cols, vals = appendIfNonNil(cols, vals, "r30_users_android", sr.Common.R30UsersAndroid)
+	cols, vals = appendIfNonNil(cols, vals, "r30_users_ios", sr.Common.R30UsersIOS)
+	cols, vals = appendIfNonNil(cols, vals, "r30_users_electron", sr.Common.R30UsersElectron)
+	cols, vals = appendIfNonNil(cols, vals, "r30_users_web", sr.Common.R30UsersWeb)
+
+	cols, vals = appendIfNonNil(cols, vals, "r30v2_users_all", sr.Common.R30V2UsersAll)
+	cols, vals = appendIfNonNil(cols, vals, "r30v2_users_android", sr.Common.R30V2UsersAndroid)
+	cols, vals = appendIfNonNil(cols, vals, "r30v2_users_ios", sr.Common.R30V2UsersIOS)
+	cols, vals = appendIfNonNil(cols, vals, "r30v2_users_electron", sr.Common.R30V2UsersElectron)
+	cols, vals = appendIfNonNil(cols, vals, "r30v2_users_web", sr.Common.R30V2UsersWeb)
```

Comment:
Is Dendrite also calculating both of these metrics, in the same way as Synapse?

Reply:
[Dendrite](https://github.com/matrix-org/dendrite/blob/1bfe87aa5614d68cbb0cad127b375048cdc70ca9/userapi/storage/postgres/stats_table.go) is using the same queries as Synapse does, so they should be the same.

Reply:
Perfect :)